pax-chassis-web/interface/src/classes/native-element-pool.ts

```diff
@@ -346,14 +365,32 @@ export class NativeElementPool {
     textUpdate(patch: TextUpdatePatch) {
         let leaf = this.nodesLookup.get(patch.id!) as HTMLElement;
         let textChild = leaf!.firstChild as HTMLElement;
+        // should be start listening to this elements size and
+        // send interrupts to the engine, or not?
+        let start_listening = false;
+
         // Handle size_x and size_y
         if (patch.size_x != null) {
-            leaf!.style.width = patch.size_x + "px";
+
+            // if size_x = -1.0, the engine wants to know
+            // this elements size from the chassi.
+            if (patch.size_x == -1.0) {
+                start_listening = true;
+            } else {
+                leaf!.style.width = patch.size_x + "px";
+            }
         }
         if (patch.size_y != null) {
-            leaf!.style.height = patch.size_y + "px";
+            if (patch.size_y == -1.0) {
+                start_listening = true;
+            } else {
+                leaf!.style.height = patch.size_y + "px";
+            }
         }
 
+        if (start_listening) {
```

Comment:
Why do we need this boolean? Let's pass the data as an enum like discussed above and just check that.

Reply:
See above